### PR TITLE
fix: allow serverFn errors to also have context

### PR DESCRIPTION
**Pull request by tannerlinsley**

### CODE REVIEW

File: packages/start/src/client/createServerFn.ts [hunk 1]
```diff
@@ -426,33 +436,55 @@ async function executeMiddleware(
...
 
-    const middlewareFn =
+    const middlewareFn = (
       env === 'client'
         ? nextMiddleware.options.client
         : nextMiddleware.options.server
+    ) as MiddlewareFn | undefined
 
     if (middlewareFn) {
+      console.log('middlewareFn', ctx)
```
Comment (schiller-manuel) [inline packages/start/src/client/createServerFn.ts:446]: remove

**Review by schiller-manuel — CHANGES_REQUESTED** [review]
no console logs please

File: packages/start/src/client/createServerFn.ts [hunk 2]
```diff
@@ -426,33 +436,55 @@ async function executeMiddleware(
...
+        // If there is a clientAfter function and we are on the client
+        const clientAfter = nextMiddleware.options.clientAfter as
+          | MiddlewareFn
+          | undefined
+
+        if (env === 'client' && clientAfter) {
+          // We need to await the next middleware and get the result
+          const result = await next(newCtx)
+
+          console.log('newCtx', newCtx)
```
Comment (schiller-manuel) [inline packages/start/src/client/createServerFn.ts:458]: remove here as well

**Comment by tannerlinsley** [comment]
Logs gone.

**Pull request merged**